spec/util/extensions/miq-module_spec.rb

```diff
@@ -41,6 +41,30 @@
       expect(test_module.default).to eq(value)
     end
 
+    it 'clears the cache even if the block raises an exception' do
+      $call_count = 0
+      test_class.cache_with_timeout(:flapping_method) do
+        begin
+          if $call_count.even?
+            $call_count
+          else
+            raise "OOPS"
+          end
+        ensure
+          $call_count += 1
+        end
+      end
+
+      test_class.flapping_method(true)             # initializes cache to 0
+      expect(test_class.flapping_method).to eq(0)  # returns cached 0
+
+      test_class.flapping_method(true) rescue nil  # blows up, clears cache for next request
+      expect(test_class.flapping_method).to eq(2)  # sets new value since cache is cleared
```

Comment:
You might want to throw another call after this one to show that it's cached since technically this line is a fresh call.